tests/src/test/fdc3.broadcast.ts

```diff
@@ -822,36 +1119,77 @@ export default () =>
       }
     }
 
-    function resetChannelsAppContext() {
-      channelsAppContext.broadcastMultipleItems = false;
-      channelsAppContext.useAppChannel = false;
-      channelsAppContext.contextBroadcasts.contact = false;
-      channelsAppContext.reverseMethodCallOrder = false;
-      channelsAppContext.broadcastExecutionComplete = false;
-    }
-
-    const executionCompleteListener = (
+    const waitForContext = (
       contextType: string,
+      testId: string,
       channel?: Channel
     ) => {
       return new Promise<Context>(async (resolve) => {
-        if (channel === undefined) {
-          const listener = await window.fdc3.addContextListener(
-            contextType,
-            (context) => {
+        let listener: Listener | null = null;
```

Comment:
`listener` is already declared right at the top of the page.

Reply:
You're right, didn't notice that. However, I just changed it from a const to a let it was already being [redeclared in main](https://github.com/finos/FDC3-conformance-framework/blob/93cd43f0603fdc0fb6400d6631503e7d1ea1389c/tests/src/test/fdc3.broadcast.ts#L847) and [in your PR](https://github.com/Joe-Dunleavy/FDC3-conformance-framework/blob/067d1dd3951776be90091dbfdc7e186afe3ddc30/tests/src/test/fdc3.broadcast.ts#L1061)

Will fix - although I haven't checked if you needed it to NOT overwrite something else using listener...

Reply:
yep you do - check out: https://github.com/finos/FDC3-conformance-framework/blob/46a7f1fb1085c8e59f2220fa8bb77c838ef28d1b/tests/src/test/fdc3.broadcast.ts#L42-L50

Thats two overlapping uses of `listener`, one through waitForContext, the other in the test definition...

Reply:
adding in an 'executionListener' to use in WaitForContext, which is (I bleive) only used for messages relating to executionComplete and windowClosed (although those must also not clash - but I think safe as the close version is only used in `afterEach`, right?).

Reply:
Yes windowClosed and closeWindow are only used in afterEach